src/gui/mzroll/videoplayer.cpp

```diff
@@ -0,0 +1,162 @@
+#include "videoplayer.h"
+
+#include <QtWidgets>
+#include <QVideoWidget>
+#include <QSizePolicy>
+#include <QSettings>
+#include <QPushButton>
+
+VideoPlayer::VideoPlayer(QSettings *settings,QWidget* parent)
+    : QWidget(parent)
+    , m_settings(settings)
+{
+    resize(750,700);
+    QBoxLayout *layout = new QVBoxLayout(this);
+    layout->setContentsMargins(20,20,20,20);
+    m_title = new QLabel(this);
+    m_title->setText("<b>Make your analyses more insightful with ML. View your fluxomics workflow in PollyPhi</b>");
+    m_title->setContentsMargins(0,0,0,10);
+    m_title->setAlignment(Qt::AlignCenter);
+    layout->addWidget(m_title);
+
+    m_message = new QLabel(this);
+    m_message->setWordWrap(true);
+    m_message->setAlignment(Qt::AlignCenter);
+    m_message->setContentsMargins(0,0,0,20);
+    m_message->setText("Classify groups as Good and Bad quickly and with accuracy with the new Machine Learning \
+algorithm. Save time classifying data and spend more time analyzing it. ");
+    layout->addWidget(m_message);
+
+
+    m_extraText =new QLabel(this);
+    m_extraText->setText("To know more view the demo below");
+    m_extraText->setAlignment(Qt::AlignCenter);
+
+    layout->addWidget(m_extraText);
+
+    m_vidWidget = new QVideoWidget(this);
+    m_vidWidget->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::MinimumExpanding);
+    m_vidWidget->setContentsMargins(0, 0, 0, 0);
+    layout->addWidget(m_vidWidget);
+
+
+    layout->itemAt(0)->setAlignment(Qt::AlignCenter);
+
+
+    m_mediaPlayer = new QMediaPlayer(this, QMediaPlayer::VideoSurface);
+    m_mediaPlayer->setVideoOutput(m_vidWidget);
+
+
+    QBoxLayout* controlsLayout = new QHBoxLayout(this);
+
+    m_playButton = new QPushButton;
+    m_playButton->setEnabled(false);
+    m_playButton->setIcon(style()->standardIcon(QStyle::SP_MediaPlay));
+
+    controlsLayout->addWidget(m_playButton);
+
+    m_positionSlider = new QSlider(Qt::Horizontal);
+    m_positionSlider->setRange(0, 0);
+    controlsLayout->addWidget(m_positionSlider);
+
+    layout->addLayout(controlsLayout);
+
+    QBoxLayout* footerLayout = new QHBoxLayout(this);
+
+    m_checkBox = new QCheckBox(this);
+    m_checkBox->setText("Don't show this again");
+    m_checkBox->setChecked(m_settings->value("hideVideoPlayer", 0).toBool());
+    footerLayout->addWidget(m_checkBox);
+
+    m_closeButton = new QPushButton(this);
+    m_closeButton->setText("Close");
+
+    footerLayout->addWidget(m_closeButton);
+    footerLayout->itemAt(1)->setAlignment(Qt::AlignRight);
+
+    layout->addLayout(footerLayout);
+
+
+    connect(m_playButton, &QAbstractButton::clicked,
+            this, &VideoPlayer::play);
+    connect(m_positionSlider, &QAbstractSlider::sliderMoved,
+            this, static_cast<void (VideoPlayer::*)(int)>(&VideoPlayer::setPosition));

```

Comment:
This is a peculiar cast. I have never seen it. I could not find such an example just by searching around. What is it doing? (just curious :)

Reply:
In case of overloaded functions we need to let the object of that class know which overloaded function to call. I am also unsure what exactly the syntax means but broadly speaking , in this case it's placing the address of the function which takes integer as one the arguments . More [here](https://wiki.qt.io/New_Signal_Slot_Syntax#Overload)